path_config.go

```diff
@@ -188,9 +210,47 @@ func (b *azureSecretBackend) pathConfigWrite(ctx context.Context, req *logical.R
 		return logical.ErrorResponse(merr.Error()), nil
 	}
 
+	// set up rotation after everything is fine
+	var rotOp string
+	if config.ShouldDeregisterRotationJob() {
+		rotOp = "deregistration"
+		// Ensure de-registering only occurs on updates and if
+		// a credential has actually been registered (rotation_period or rotation_schedule is set)
+		deregisterReq := &rotation.RotationJobDeregisterRequest{
+			MountPoint: req.MountPoint,
+			ReqPath:    req.Path,
+		}
+		if prevRotateCfg {
+			err := b.System().DeregisterRotationJob(ctx, deregisterReq)
+			if err != nil {
+				return logical.ErrorResponse("error de-registering rotation job: %s", err), nil
+			}
+		}
+	} else if config.ShouldRegisterRotationJob() {
+		rotOp = "registeration"
+		req := &rotation.RotationJobConfigureRequest{
+			Name:             rootRotationJobName,
+			MountPoint:       req.MountPoint,
+			ReqPath:          req.Path,
+			RotationSchedule: config.RotationSchedule,
+			RotationWindow:   config.RotationWindow,
+			RotationPeriod:   config.RotationPeriod,
+		}
+
+		_, err := b.System().RegisterRotationJob(ctx, req)
+		if err != nil {
+			return logical.ErrorResponse("error registering rotation job: %s", err), nil
+		}
+	}
+
 	err = b.saveConfig(ctx, config, req.Storage)
 	if err != nil {
-		return nil, err
+		wrappedError := err
```

Comment:
can we add the log back in since we're doing both the log and the error in the other plugins?